modules/csvparser/csvparser.c

```diff
@@ -165,6 +166,81 @@ dispatch_key_formatter(gchar *prefix)
   return prefix ? _format_key_for_prefix : _return_key;
 }
 
+static gboolean
+_should_drop_message(CSVParser *self)
+{
+  return self->drop_invalid || (self->template_options.on_error & ON_ERROR_DROP_MESSAGE);
+}
+
+static gboolean
+_process_column_on_invalid_type_cast(CSVParser *self, gchar *current_column_name,
+                                     LogMessageValueType *current_type, GError *error)
+{
+  if(self->template_options.on_error & ON_ERROR_FALLBACK_TO_STRING)
+    {
+      if (!(self->template_options.on_error & ON_ERROR_SILENT))
+        {
+          msg_debug("csv-parser: error casting value to the type specified, trying string type because on-error(\"fallback-to-string\") was specified",
+                    evt_tag_str("column", current_column_name),
+                    evt_tag_str("type", log_msg_value_type_to_str(*current_type)),
+                    evt_tag_str("error", error->message));
+        }
+      *current_type = LM_VT_STRING;
+      return TRUE;
+    }
+  else if (self->template_options.on_error & ON_ERROR_DROP_PROPERTY)
+    {
+      if (!(self->template_options.on_error & ON_ERROR_SILENT))
+        {
+          msg_debug("csv-parser: error casting value to the type specified, dropping property because on-error(\"drop-property\") was specified",
+                    evt_tag_str("column", current_column_name),
+                    evt_tag_str("type", log_msg_value_type_to_str(*current_type)),
+                    evt_tag_str("error", error->message));
+        }
+    }
+  return FALSE;
+}
+
+
+static gboolean
+_process_column(CSVParser *self, CSVScanner *scanner, LogMessage *msg, GList *column_l, GString *key_scratch)
+{
+  CSVParserColumn *current_column = column_l->data;
+  LogMessageValueType current_column_type = current_column->type;
+  const gchar *current_value = csv_scanner_get_current_value(scanner);
+  key_formatter_t _key_formatter = dispatch_key_formatter(self->prefix);
+
+  GError *error = NULL;
+  gboolean should_set_value = TRUE;
+
+  if (!type_cast_validate(current_value, current_column_type, &error))
+    {
+      if (_should_drop_message(self))
```

Comment:
Shouldn't we check whether `silently` was set?

Reply:
Yeah, nice catch, added it in the latest revision!